src/opnsense/mvc/app/controllers/OPNsense/CaptivePortal/Api/AccessController.php

```diff
@@ -102,6 +100,25 @@ private function getClientIp()
         }
     }
 
+    private function getClientMac($ip)
+    {
+        $mac = null;
+
+        if (empty(self::$arp)) {
+            self::$arp = json_decode((new Backend())->configdRun("interface list arp json"), true);
+        }
+
+        if (self::$arp != null) {
+            foreach (self::$arp as $arp) {
+                if (!empty($arp['ip'] && $arp['ip'] == $ip)) {
+                    $mac = $arp['mac'];
```

Comment:
may short circuit here (`return $arp['mac']`) in which case we can remove `$mac`